.github/workflows/main.yml

```diff
@@ -267,8 +267,8 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
 
-      - name: Use conda for group1 and group2
-        if: matrix.test-suite == 'group1' || matrix.test-suite == 'group2'
+      - name: Use conda for group1 and group3
```

Comment:
Looking at the code below, as these groups dynamic?

Reply:
Not sure what you mean by dynamic. The name? No. The number of them? No. There's always 4. Although it's not decided here. runFunctionalTests.js decides the number.

Reply:
It would probably be better to actually set the groups in the yml file than in some javascript file. Although writing code in bash scripts is not my forte. But then groups would be decided in the same spot as we're using them.

Reply:
I'll enter an issue for that.

Reply:
Sorry, not exactly what I meant. I looks like the grouping code is making the bucket of files for each group based on file size. So if the size of files changes then something that was in group X might now be in group Y, which would mess up which tests were running with conda. Is that not the case?

Reply:
Yes but that was kinda on purpose. I wanted it to be random which tests used conda as they should all work. 

However for those that need to be a certain group, I added the logic for forcing it. 

Not super happy with it all because it's so separated. I think the issue I entered would make it better.

Reply:
Do you think we should 
- Force certain tests to run with conda?
or
- Run all tests with conda and without?

I was thinking we could do that too but didn't think it mattered that much. Conda failures generally happen for everything the same way when they do happen, so just running some of the tests with conda would verify conda works.

Reply:
Yeah, I see the forcing now. The tricky bit is that if random tests are assigned to conda then we get run to run variability which always seems like a bad thing in general. Getting test randomly assigned to conda would throw off David's perf tests as well I would think. Lemme look at the new issue.